apps/vsce/src/components/engineService.ts

```diff
@@ -228,30 +230,43 @@ export class EngineService {
 		messageBus.subscribe(MessageKind.executeCodemodSet, (message) => {
 			this.#onExecuteCodemodSetMessage(message);
 		});
+
+		this.__pollCodemodEngineNode();
 	}
 
-	async #onEnginesBootstrappedMessage(
-		message: Message & { kind: MessageKind.engineBootstrapped },
-	) {
-		this.__codemodEngineNodeExecutableUri =
-			message.codemodEngineNodeExecutableUri;
-		this.__codemodEngineRustExecutableUri =
-			message.codemodEngineRustExecutableUri;
+	private async __pollCodemodEngineNode() {
+		const checkCodemodEngineNode = async () => {
+			const codemodEngineNodeLocated =
+				await this.isCodemodEngineNodeLocated();
 
-		await this.syncRegistry();
+			if (codemodEngineNodeLocated) {
+				this.#messageBus.publish({
+					kind: MessageKind.codemodEngineNodeLocated,
+				});
+				this.__onCodemodEngineNodeLocated();
+				clearInterval(codemodEnginePollingIntervalId);
+			}
+		};
+
+		checkCodemodEngineNode();
+		// we retry codemod engine installation checks automatically, so we can detect when user installs the codemod
+		const codemodEnginePollingIntervalId = setInterval(
```

Comment:
when is this interval cancelled?

Reply:
Its cancelled when we successfully locate the engine, in the checkCodemodEngineNode function

Reply:
Wow, I would assume TS would comment on such an usage of the interval, can we constraint it as well to the max number of checks (e.g. 10?)